base/src/types.rs

```diff
@@ -657,7 +648,7 @@ impl<Id> ASTType<Id> {
     /// Returns the inner most application of a type application
     pub fn inner_app(&self) -> &ASTType<Id> {
         match **self {
-            Type::App(ref a, _) => a.inner_app(),
+            Type::Data(ref a, _) => a.inner_app(),
```

Comment:
I suspect the field access panic only happens because it tries to instantiate `Monoid` rather than `Monoid a`due to the Inner_app call. It worked before first because the compiler had its own way of looking through the aliases which I changed yesterday to use `instantiate::remove_aliases `.  But since Inner_app didn't handle `Data` it still worked until now.